### PR TITLE
Add test case cpu_count_check.

### PR DESCRIPTION
**Pull request by LiliDeng**
```
2021-09-05 10:27:26.483[10764][INFO] lisa.RootRunner                                CPU.cpu_count_check: PASSED   
```

### CODE REVIEW

File: lisa/tools/lscpu.py [hunk 1]
```diff
@@ -104,6 +107,36 @@ def get_core_count(self, force_run: bool = False) -> int:
 
         return self._core_count
 
+    def get_thread_per_core_count(self, force_run: bool = False) -> int:
```
Comment (squirrelsc) [inline lisa/tools/lscpu.py:110]: What's the reason it needs the force run? The value should be fixed, is it right?

Comment (LiliDeng) [reply lisa/tools/lscpu.py:110]: if it is used in resize vm size, then it will be different.

File: microsoft/testsuites/core/cpu.py [hunk 2]
```diff
@@ -148,3 +148,32 @@ def l3_cache_check(self, node: Node, log: Logger) -> None:
                 "L3 cache of each core must be mapped to the NUMA node "
                 "associated with the core.",
             ).is_equal_to(cpu.numa_node)
+
+    @testsuite.TestCaseMetadata(
+        description="""
+            This test will check that vCPU count correctness.
+            Steps :
+            1. Get vCPU count.
+            2. Caculate vCPU count by core_per_socket_count * socket_count *
```
Comment (dsrivastavv) [inline microsoft/testsuites/core/cpu.py]: [nit] calculate

Comment (LiliDeng) [reply microsoft/testsuites/core/cpu.py]: Fixed

**Review by squirrelsc — APPROVED** [review]

**Pull request merged**